### PR TITLE
add null checking in SerializableErrorDecoder

### PR DESCRIPTION
**Pull request by sixinli**
response can have null body (some of the 401 errors that i've seen got that)

**Goals (and why)**:
give better error

**Implementation Description (bullets)**:
add null checking for error response body - so it returns the proper error instead of throwing a NPE when body is null

**Priority (whenever / two weeks / yesterday)**:
whenever 

---
This change is [Reviewable](https://reviewable.io/reviews/palantir/atlasdb/2446)

### CODE REVIEW

File: atlasdb-feign/src/main/java/com/palantir/atlasdb/http/SerializableErrorDecoder.java [hunk 1]
```diff
@@ -48,7 +48,9 @@ public Exception decode(String methodKey, Response response) {
 
     private static InputStream getBody(Response response) {
         try {
-            return response.body().asInputStream();
+            if (response.body() != null) {
+                return response.body().asInputStream();
+            }
         } catch (IOException e) {
             log.warn("Unable to read message body from response {}", response, e);
             return null;
```
Comment (jeremyk-91) [inline atlasdb-feign/src/main/java/com/palantir/atlasdb/http/SerializableErrorDecoder.java]: We'll probably need to move this return out of the catch-block if we're going ahead with this. I'm guessing currently we get an NPE that blows up, so the change makes sense in general.

**Comment by tboam** [comment]
@sixinli can you add some release notes and then this can merge

**Comment by sixinli** [comment]
ping, any idea with the circle ci thing? cant open the link

**Comment by jeremyk-91** [comment]
@sixinli sorry about the delay, was ooto. The previous build looked like a flake, so I've kicked off another build and will merge this when that passes.

**Pull request merged**